### DISCUSSION TITLE
Horizontal menu (menubar)

### DISCUSSION THREAD

**Discussion by alekseykurylev** [adobe/react-spectrum#5624] [Q&A]
Hi! How can I make a horizontal menu with a drop-down menu using RAC components. Example https://www.w3.org/WAI/ARIA/apg/patterns/menubar/examples/menubar-navigation/

**Comment by snowystinger** [comment]
We haven't built anything expressly for that. However, I think you could achieve it with `Toolbar` + `Menu`s
https://react-spectrum.adobe.com/react-aria/Toolbar.html
https://react-spectrum.adobe.com/react-aria/Menu.html

There may be a conflict with SubMenus when they are eventually supported.

**Reply by haidarzhy** [reply]
Hi!
@snowystinger Since submenus are officially released, is there correct way to implement such menu?

I've tried to put top-level horizontal menu in just Menu, but it didn't work correctly without Menu trigger.

It's more or less work if top level menus are separate Menu components with each own triggers, but it's missing key feature from @alekseykurylev link:
- if I click on one of submenus and move mouse to the neighbor submenu, popover is closed, and neighbor popover is open automatically

![Recording 2024-05-17 at 11 21 43](https://github.com/adobe/react-spectrum/assets/576394/9b3fe441-256b-47c1-b046-a36b94ad08fc)

**Reply by snowystinger** [reply]
We've not tried to build this yet.
Linking together related Issues/Discussions https://github.com/adobe/react-spectrum/discussions/6441#discussioncomment-9542815
